cmd/security-agent/main_windows.go

```diff
@@ -0,0 +1,176 @@
+	_ "net/http/pprof"
+	"os"
+	"path/filepath"
+
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/security-agent/app"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	_ "github.com/DataDog/datadog-agent/pkg/util/containers/providers/windows"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil"
+
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/eventlog"
+
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+)
+
+var (
+	elog           debug.Log
+	defaultLogFile = "c:\\programdata\\datadog\\logs\\security-agent.log"
+
+	// DefaultConfPath points to the folder containing datadog.yaml
+	DefaultConfPath = "c:\\programdata\\datadog"
+
+	enabledVals = map[string]bool{"yes": true, "true": true, "1": true,
+		"no": false, "false": false, "0": false}
+	// MIKEZHU: Questionable
+	subServices = map[string]string{"logs_enabled": "logs_enabled",
+		"apm_enabled":     "apm_config.enabled",
+		"process_enabled": "process_config.enabled"}
+)
+
+func init() {
+	// MIKEZHU: We need to make change to the MSI to install Security Agent.
+	pd, err := winutil.GetProgramDataDirForProduct("Datadog Security Agent")
+	if err == nil {
+		DefaultConfPath = pd
+		defaultLogFile = filepath.Join(pd, "logs", "security-agent.log")
+	} else {
+		winutil.LogEventViewer(ServiceName, 0x8000000F, defaultLogFile)
+	}
+}
+
+// ServiceName is the name of the service in service control manager
+const ServiceName = "datadog-security-agent"
+
+// EnableLoggingToFile -- set up logging to file
+
+func main() {
+	// set the Agent flavor
+	flavor.SetFlavor(flavor.Dogstatsd)
+	config.Datadog.AddConfigPath(DefaultConfPath)
+
+	// go_expvar server
+	go func() {
+		var port = coreconfig.Datadog.GetString("security_agent.expvar_port")
+		coreconfig.Datadog.Set("expvar_port", port)
+		if coreconfig.Datadog.GetBool("telemetry.enabled") {
+			http.Handle("/telemetry", telemetry.Handler())
+		}
+		err := http.ListenAndServe("127.0.0.1:"+port, http.DefaultServeMux)
+		if err != nil && err != http.ErrServerClosed {
+			log.Errorf("Error creating expvar server on port %v: %v", port, err)
+		}
+	}()
+
+	isIntSess, err := svc.IsAnInteractiveSession()
+	if err != nil {
+		fmt.Printf("failed to determine if we are running in an interactive session: %v\n", err)
+	}
+	if !isIntSess {
+		// confPath = DefaultConfPath
+		runService(false)
+		return
+	}
+	defer log.Flush()
+
+	if err = app.SecurityAgentCmd.Execute(); err != nil {
+		log.Error(err)
+		os.Exit(-1)
+	}
+}
+
+type myservice struct{}
+
+func (m *myservice) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	changes <- svc.Status{State: svc.StartPending}
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
```

Comment:
I've created a new PR because the merge from master to this branch went wrong...

The new PR is https://github.com/DataDog/datadog-agent/pull/7791

The issue has been fixed .